libsolidity/analysis/TypeChecker.cpp

```diff
@@ -39,17 +39,13 @@ namespace
 
 bool typeSupportedByOldABIEncoder(Type const& _type)
 {
-	if (_type.dataStoredIn(DataLocation::Storage))
-		return true;
-	else if (_type.category() == Type::Category::Struct)
+	if (_type.category() == Type::Category::Struct)
```

Comment:
This changes semantics - previously a storage struct returned true, but now it returns fals.

Reply:
Uha.... Understood @chriseth. :+1: It took a little bit time, but finally got the issue here. Anyway, I'm push another commit fixing this. :slightly_smiling_face: